crates/cranelift/src/compiler/component.rs

```diff
@@ -46,6 +46,41 @@ impl ComponentCompiler for Compiler {
             self.wasm_to_host_spill_args(ty, &mut builder, block0);
         let vmctx = builder.func.dfg.block_params(block0)[0];
 
+        // Save the exit FP and return address for stack walking purposes.
+        //
+        // First we need to get the `VMRuntimeLimits`.
+        let limits = builder.ins().load(
+            pointer_type,
+            MemFlags::trusted(),
+            vmctx,
+            i32::try_from(offsets.limits()).unwrap(),
+        );
+        // Then save the exit Wasm FP to the limits.
```

Comment:
Could this grow a comment explaining why we're reading our prior fp as opposed to using the fp for this trampoline?